### PR TITLE
Android: DrawerLayout

### PR DESCRIPTION
**Pull request by m1ga**
Adding Android DrawerLayout example

~~* either wait for https://jira.appcelerator.org/browse/AC-5196~~
~~* or add `<property name="run-on-main-thread" type="bool">true</property>` to tiapp.xml~~

added `<property name="run-on-main-thread" type="bool">true</property>` to tiapp.xml

### CODE REVIEW

**Comment by hansemannn** [comment]
Very ok with adding the property, as it should be default anyway these days.

**Comment by m1ga** [comment]
👍 ok, then it will run. Added main-thread

File: app/controllers/phone/drawer.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+$.btn_left.addEventListener('click', function(e) {
+	$.drawerLayout.toggleLeft();
+});
+$.btn_right.addEventListener('click', function(e) {
+	$.drawerLayout.toggleRight();
+});
+$.close.addEventListener("click", function(e) {
+	$.drawer.close();
+})
+$.drawer.addEventListener('open', function() {
```
Comment (hansemannn) [inline app/controllers/phone/drawer.js]: Why not do this via XML event-listeners?

File: app/controllers/phone/index.js [hunk 2]
```diff
@@ -11,7 +11,9 @@ function openComponent(e) {
     var identifier = 'phone/' + e.section.getItemAt(e.itemIndex).properties.itemId;
     var component = Alloy.createController(identifier).getView();
     
-    Alloy.Globals.setAndroidBackButton(component);
+    if (OS_ANDROID && identifier != "phone/drawer") {
```
Comment (hansemannn) [inline app/controllers/phone/index.js]: use !==

File: app/styles/phone/drawer.tss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+"#btn_left" : {
+	top: 100
+}
+"#btn_right" : {
+	bottom: 100
+}
+
+".views": {
```
Comment (hansemannn) [inline app/styles/phone/drawer.tss]: I prefer `'` vs `"` for all quotes.

**Comment by m1ga** [comment]
Made the adjustments 👍

**Comment by hansemannn** [comment]
Beautiful! Approved 🚀

**Pull request merged**